2021/day20/day20.java

```diff
@@ -0,0 +1,108 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.File;
+
+public class day20 {
+    public static String readAlgorithm(String filename) {
+        String algorithm = null;
+        try {
+            Scanner sc = new Scanner(new File(filename));
+            algorithm = sc.nextLine();
+            sc.close();
+        } catch (Exception e) {
+            System.out.println("Something went wrong " + e);
+        }
+        return algorithm;
+    }
+
+    public static ArrayList<ArrayList<Character>> readImage(String filename) {
+        ArrayList<ArrayList<Character>> image = new ArrayList<>();
+        try {
+            Scanner sc = new Scanner(new File(filename));
+            sc.nextLine();
+            sc.nextLine();
+            sc.useDelimiter("");
+            while (sc.hasNextLine()) {
+                image.add(stringToList(sc.nextLine()));
+            }
+            sc.close();
+        } catch (Exception e) {
+            System.out.println("Something went wrong " + e);
+        }
+        return image;
+    }
+
+    public static ArrayList<ArrayList<Character>> surroundWithDots(ArrayList<ArrayList<Character>> image) {
+        ArrayList<ArrayList<Character>> newImage = new ArrayList<>();
+        newImage.add(new ArrayList<>());
```

Comment:
This should be replaced with `new ArrayList<>(Collections.nCopies(image.get(0).size() + 2), '.')`